test/NuGet.Core.Tests/NuGet.PackageManagement.Test/PackageDownloaderTests.cs

```diff
@@ -623,6 +624,32 @@ public async Task GetDownloadResourceResultAsync_SupportsDownloadResultWithoutPa
             }
         }
 
+        [Fact]
+        public async Task GetDownloadResourceResultAsync_InvalidPackageId_Throws()
+        {
+            // Arrange
+            using TestDirectory testDirectory = TestDirectory.Create();
+            using FileSystemBackedV3MockServer server = new FileSystemBackedV3MockServer(testDirectory.Path);
+            server.Start();
+            PackageSource packageSource = new PackageSource(server.ServiceIndexUri, "http-source")
+            {
+                AllowInsecureConnections = true
+            };
+            var repo = Repository.Factory.GetCoreV3(packageSource);
+            var downloadResource = await repo.GetResourceAsync<DownloadResource>(CancellationToken.None);
+            string id = "../contoso";
+
+            // Act & Assert
+            var exception = await Assert.ThrowsAsync<InvalidPackageIdException>(() => downloadResource.GetDownloadResourceResultAsync(
+                new PackageIdentity(id, NuGetVersion.Parse("1.0.0")),
+                new PackageDownloadContext(new SourceCacheContext()),
+                "",
+                NullLogger.Instance,
+                CancellationToken.None));
+            server.Stop();
+            exception.Message.Should().Contain(string.Format("Invalid package id : `{0}`", id));
```

Comment:
The goal of not using hard-coded strings in tests where possible remains, but in this case, due to circular project references, we agreed it's out of scope for this PR. By the way, I hit the same issue before and addressed some circular references in https://github.com/NuGet/NuGet.Client/pull/5052